shared/hooks/useComponentSize.ts

```diff
@@ -63,5 +55,15 @@ export function useComponentSize(
     };
   });
 
+  useLayoutEffect(() => {
```

Comment:
Moved this afterwards as it makes sense to setup the listeners _after_ the handlers.